src/Microsoft.DocAsCode.Build.Engine/TemplateManager.cs

```diff
@@ -36,6 +37,20 @@ public TemplateProcessor GetTemplateProcessor(DocumentBuildContext context, int
             return new TemplateProcessor(new CompositeResourceCollectionWithOverridden(_templates.Select(s => _finder.Find(s)).Where(s => s != null)), context, maxParallelism);
         }
 
+        public string GetTemplatesHash()
+        {
+            using (var ms = new MemoryStream())
+            using (var writer = new StreamWriter(ms))
+            {
+                var resources = _templates.Select(s => _finder.Find(s)).Where(s => s != null);
+                string content = string.Concat(from r in resources
+                                               from n in r.Names
+                                               select r.GetResource(n));
```

Comment:
nice!

---
In reply to: [75412827](https://github.com/dotnet/docfx/pull/621#discussion_r75412827) [](ancestors = 75412827)